### PR TITLE
Velodyne Cloud Accumulate

### PR DESCRIPTION
**Pull request by gtinchev**
Added velodyne accumulator corresponding to bot_core::pointcloud_t message.

### CODE REVIEW

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp [hunk 1]
```diff
@@ -210,3 +211,76 @@ bool CloudAccumulate::processVelodyne(const bot_core::pointcloud2_t* msg){
 
   return true;
 }
+
+bool CloudAccumulate::processVelodynePointcloud(bot_core::pointcloud_t* msg){
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: please call these functions processPointcloud. Any data of type pointcloud_t could be handled by this function.

also these intermediate functions shouldn't be necessary. (it wasn't needed for processVelodyne)

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp [hunk 2]
```diff
@@ -59,6 +60,10 @@ class CloudAccumulate{
 
     bool processVelodyne(const bot_core::pointcloud2_t* msg);
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.hpp]: rename this function to processPointcloud2 (to match other one).
they should be generic eventually

File: motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp [hunk 3]
```diff
@@ -210,3 +211,76 @@ bool CloudAccumulate::processVelodyne(const bot_core::pointcloud2_t* msg){
...
+  pronto::PointCloud* scan_velodyne (new pronto::PointCloud ());
+  size_t npts = cloud->points.size();
+  scan_velodyne->points.resize(npts);
+  for(size_t j=0; j<npts; ++j) {
+    scan_velodyne->points[j].x = cloud->points[j].x;
+    scan_velodyne->points[j].y = cloud->points[j].y;
+    scan_velodyne->points[j].z = cloud->points[j].z;
+  }
+
+  // 4. Project the scan into local frame:
```
Comment (mauricefallon) [inline motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: could you reuse everything from here to end from processVelodyne() - after you have converted the data into a pronto::Pointcloud*   ?

Comment (gtinchev) [reply motion_estimate/src/cloud_accumulate/cloud_accumulate.cpp]: I would like to keep the function processPointcloud that accepts a PCL pointcloud, as I'm using it directly from my own code.
Perhaps I'll rename it from processPointcloud to processPCLPointcloud, and processVelodynePointcloud to processPointcloud (accepting bot_core::pointcloud_t).

**Comment by gtinchev** [comment]
@mauricefallon could you please review and merge?

**Comment by mauricefallon** [comment]
That's fine, thanks!

**Pull request merged**